xds/src/test/java/io/grpc/xds/XdsClientImplTest.java

```diff
@@ -1212,6 +1218,37 @@ public void cdsResponseWithMatchingResource() {
     assertThat(clusterUpdate.getLrsServerName()).isEqualTo("");
   }
 
+  /**
+   * CDS response containing UpstreamTlsContext for a cluster.
+   */
+  @Test
+  public void cdsResponseWithUpstreamTlsContext() {
+    xdsClient.watchClusterData("cluster-foo.googleapis.com", clusterWatcher);
+    StreamObserver<DiscoveryResponse> responseObserver = responseObservers.poll();
+    StreamObserver<DiscoveryRequest> requestObserver = requestObservers.poll();
+
+    // Management server sends back CDS response with UpstreamTlsContext.
+    UpstreamTlsContext testUpstreamTlsContext =
+            buildUpstreamTlsContext("secret1", "unix:/var/uds2");
+    List<Any> clusters = ImmutableList.of(
```

Comment:
FYI: `ImmutableList` should be stored in `ImmutableList` otherwise any mutable operation will throw exception. not really applicable here. hence, it is fyi.

Reply:
Here I think it only uses `ImmutableList` as a convenient way to build some proto message, so it should be fine.

Reply:
Not making any changes.